pylint/checkers/base/basic_checker.py

```diff
@@ -266,6 +266,12 @@ class BasicChecker(_BasicChecker):
             "Used when an exception is created without being assigned, raised or returned "
             "for subsequent use elsewhere.",
         ),
+        "W0134": (
+            "'%s' shadowed by the 'finally' clause.",
+            "return-or-yield-in-finally",
+            "Emitted when a 'return' or 'yield' statement is found in a 'finally' block. "
+            "This will overwrite the return value of a function and should be avoided.",
```

Comment:
Speaking a bit out of my depth here, but I think if `return`ing from a generator, the return value won't be overwritten strictly speaking, it will just be passed to a coroutine. So this is slightly incorrectly described for `yield`s, leaving me to suggest to back out all the yield-related stuff.